engine/table/src/main/java/io/deephaven/engine/table/impl/sources/regioned/RegionedColumnSourceManager.java

```diff
@@ -893,4 +904,59 @@ public Map<String, String> renameMap(final WhereFilter filter, final ColumnSourc
     public PushdownFilterContext makePushdownFilterContext() {
         return new BasePushdownFilterContext();
     }
+
+    public static final class Builder implements Closeable {
+
+        private final PushdownResult[] results;
+
+        private long matchCount;
+        private long maybeMatchCount;
+        private long notMatchCount;
+
+        private Builder(int numResults) {
+            results = new PushdownResult[numResults];
+            matchCount = 0;
+            maybeMatchCount = 0;
+            notMatchCount = 0;
+        }
+
+        public void add(final int ix, final long selectionSize, final PushdownResult results) {
+            final long matchCount = results.match().isEmpty() ? 0 : results.match().size();
+            final long maybeMatchCount = results.maybeMatch().isEmpty() ? 0 : results.maybeMatch().size();
+            if (matchCount + maybeMatchCount > selectionSize) {
+                throw new IllegalStateException(
+                        String.format("matchCount + maybeMatchCount > selectionSize; %d + %d > %d", matchCount,
+                                maybeMatchCount, selectionSize));
+            }
+            final long notMatchCount = selectionSize - matchCount - maybeMatchCount;
+            final PushdownResult copy = results.copy();
+            synchronized (this) {
+                this.results[ix] = copy;
+                this.matchCount += matchCount;
+                this.maybeMatchCount += maybeMatchCount;
+                this.notMatchCount += notMatchCount;
+            }
+        }
+
+        public synchronized PushdownResult build(final RowSet selection) {
```

Comment:
Not a strong opinion, but maybe pass `long selectionSize` instead of the rowset so there is no confusion about ownership / cleanup.